### PR TITLE
Feature/rb 63 basic ennemy

### PR DESCRIPTION
**Pull request by guillaumeAbel**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [x] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: assets/Json/ennemyData.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                        "width": 30.0,
+                        "height": 29.0
+                    }
+                ],
+                "attack": [],
+                "dead": []
+            }
+        }
+    ]
+}
```
Comment (Saverio976) [inline assets/Json/ennemyData.json]:
```suggestion
}

```

File: src/ECS/SparseArray.hpp [hunk 2]
```diff
@@ -10,6 +10,8 @@
 #include <iterator>
 #include <vector>
 
+#include "ECSCustomTypes.hpp"
```
Comment (Saverio976) [inline src/ECS/SparseArray.hpp]:
```suggestion
```

File: src/ECS/Systems/Managers/SystemManager.cpp [hunk 3]
```diff
@@ -29,32 +30,35 @@ namespace Systems {
...
         }
-        std::size_t decrease = 0;
         std::sort(_toRemove.begin(), _toRemove.end());
         for (auto &id : _toRemove) {
             auto it = _modifiedSystems.begin();
-            std::advance(it, id - decrease);
-            _modifiedSystems.erase(it);
-            decrease++;
+            std::advance(it, id);
+            _modifiedSystems.erase(it - decrease);
```
Comment (Saverio976) [inline src/ECS/Systems/Managers/SystemManager.cpp]:
```suggestion
            _modifiedSystems.erase(it);
```

File: src/ECS/Systems/Managers/SystemManager.cpp [hunk 4]
```diff
@@ -29,32 +30,35 @@ namespace Systems {
...
             i++;
         }
-        std::size_t decrease = 0;
         std::sort(_toRemove.begin(), _toRemove.end());
         for (auto &id : _toRemove) {
             auto it = _modifiedSystems.begin();
-            std::advance(it, id - decrease);
-            _modifiedSystems.erase(it);
-            decrease++;
+            std::advance(it, id);
```
Comment (Saverio976) [inline src/ECS/Systems/Managers/SystemManager.cpp]:
```suggestion
            std::advance(it, id  - decrease);
```

Comment (Saverio976) [reply src/ECS/Systems/Managers/SystemManager.cpp]: c'est plus propre

File: src/ECS/Systems/Managers/SystemManager.cpp (hunk 3 above)
Comment (Saverio976) [reply src/ECS/Systems/Managers/SystemManager.cpp]: c'est plus propre

**Review by KitetsuK — APPROVED** [review]

File: src/ECS/Systems/Systems.cpp [hunk 5]
```diff
@@ -224,34 +221,95 @@ namespace Systems {
...
+        nlohmann::json attackData   = animRectData["attack"];
+        nlohmann::json deadData     = animRectData["dead"];
+        Registry::getInstance().getComponents<Types::AnimRect>().insertBack(Types::AnimRect(
+            Types::Rect(ennemyData["rect"]),
+            moveData.get<std::vector<Types::Rect>>(),
+            attackData.get<std::vector<Types::Rect>>(),
+            deadData.get<std::vector<Types::Rect>>()));
+        Registry::getInstance().getComponents<Types::Velocity>().insertBack(
+            {Types::Velocity(ennemyData["velocity"])});
+        Registry::getInstance().getComponents<health_s>().insertBack({ennemyData["health"]});
```
Comment (Saverio976) [inline src/ECS/Systems/Systems.cpp]: struct healt_s

**Review by Saverio976 — CHANGES_REQUESTED** [review]

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**